### PR TITLE
IDENTITY-6026: Fix retrieving user claims from cache always assume token, even when authorization code is given.

### PR DESCRIPTION
**Pull request by mefarazath**
https://wso2.org/jira/browse/IDENTITY-6026

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java [hunk 1]
```diff
@@ -185,13 +185,15 @@ public void handleCustomClaims(JWTClaimsSet jwtClaimsSet, OAuthAuthzReqMessageCo
     private Map<String, Object> getResponse(OAuthTokenReqMessageContext requestMsgCtx)
             throws OAuthSystemException {
 
+        // get the user claims cached against the access token if any
         Map<ClaimMapping, String> userAttributes =
-                getUserAttributesFromCache(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
+                getUserAttributesFromCacheUsingToken(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
```
Comment (Pushpalanka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java]: Needs a null check similar to 'requestMsgCtx.getProperty(OAuthConstants.AUTHZ_CODE) != null' here as well.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java [hunk 2]
```diff
@@ -215,7 +217,7 @@ public void handleCustomClaims(JWTClaimsSet jwtClaimsSet, OAuthAuthzReqMessageCo
             throws OAuthSystemException {
 
         Map<ClaimMapping, String> userAttributes =
-                getUserAttributesFromCache(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
+                getUserAttributesFromCacheUsingToken(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
```
Comment (Pushpalanka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java]: Check for null before calling toString.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java (hunk 1 above)
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java]: done.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java (hunk 2 above)
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java]: done.

**Pull request merged**